grammars/css.cson

```diff
@@ -336,176 +1219,377 @@
+      (?=
+        \\s*
+        (?: &             # Direct nesting
+          | @nest         # Nesting @-rule (complex nesting)
+            (?![\\w-])
+        )
+      )
+    '''
+    'end': '(?<=})\\s*|(?=[/;])'
+    'patterns': [
+      {
+        # Selector component
+        'begin': '\\G'
+        'end': '\\s*(?={)|(?=[/;])'
+        'name': 'meta.nested.selector.css'
+        'patterns': [
+          {
+            'include': '#selector-innards'
+          }
+        ]
+      }
+      {
+        'begin': '{'
+        'beginCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.begin.css'
+        'end': '}'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.end.css'
+        'name': 'meta.nested.property-list.css'
+        'patterns': [
+          {
+            'include': '#rule-list-innards'
+          }
+        ]
+      }
+    ]
+
   'numeric-values':
     'patterns': [
       {
         'captures':
           '1':
             'name': 'punctuation.definition.constant.css'
-        'match': '(#)([0-9a-fA-F]{3}|[0-9a-fA-F]{6})\\b'
-        'name': 'constant.other.color.rgb-value.css'
+        'match': '(#)(?:[0-9a-fA-F]{3,4}|[0-9a-fA-F]{6}|[0-9a-fA-F]{8})\\b'
+        'name': 'constant.other.color.rgb-value.hex.css'
       }
       {
         'captures':
           '1':
-            'name': 'keyword.other.unit.css'
-        'match': '(?x)
-          (?<!\\w|-)(?:(?:-|\\+)?(?:[0-9]+(?:\\.[0-9]+)?)|(?:\\.[0-9]+))
-          ((?:px|pt|ch|cm|mm|in|r?em|ex|pc|deg|g?rad|dpi|dpcm|dppx|fr|ms|s|turn|vh|vmax|vmin|vw)\\b|%)?
-          '
+            'name': 'keyword.other.unit.percentage.css'
+          '2':
+            'name': 'keyword.other.unit.${2:/downcase}-unit.css'
+        'match': '''(?xi) (?<!\\w|-)
+          [-+]?                               # Sign indicator
+
+          (?:                                 # Numerals
+              [0-9]+ (?:\\.[0-9]+)?           # Integer/float with leading digits
+            | \\.[0-9]+                       # Float without leading digits
+          )
+
+          (?:                                 # Scientific notation
+            (?<=[0-9])                        # Exponent must follow a digit
+            [Ee]                              # Exponent indicator
+            [-+]?                             # Possible sign indicator
+            [0-9]+                            # Exponent value
+          )?
+
+          (?:                                 # Possible unit for data-type:
+            (%)                               # - Percentage
+            | ( deg|grad|rad|turn             # - Angle
+              | Hz|kHz                        # - Frequency
+              | ch|cm|em|ex|in|mm|mozmm|pc|   # - Length
+                pt|px|q|rem|vh|vmax|vmin|vw
+              | dpi|dpcm|dppx                 # - Resolution
+              | s|ms                          # - Time
+              )\\b
```

Comment:
It's lax on purpose. Consider the possibility this rule might be embedded in Sass or Less, where spaces aren't required inside arithmetic:
```scss
a{
	width: 1em-2em;
}
```